corehq/motech/repeaters/management/commands/generate_repeater_report.py

```diff
@@ -0,0 +1,123 @@
+            dest='state',
+            help='State of Record: CANCELLED, SUCCESS, PENDING, FAIL'
+        )
+        parser.add_argument(
+            '--records_file_path',
+            dest='records_file_path',
+            help='Path to a csv file with ids under header record_id'
+        )
+
+    def _add_row(self, repeat_record):
+        row = [repeat_record.get_id, repeat_record.payload_id, repeat_record.state, repeat_record.failure_reason]
+        repeat_record_attempts_count = len(repeat_record.attempts)
+        if repeat_record_attempts_count > self.max_attempts_in_sheet:
+            self.max_attempts_in_sheet = repeat_record_attempts_count
+        for attempt in repeat_record.attempts:
+            row.append(attempt.message)
+        self.ws.append(row)
+
+    def _add_header_for_attempts(self):
+        # find first empty header cell:
+        # iterate first row till highest column length which is highest column length over the sheet
+        # Even if there are no values in header, empty cells are inserted to match highest
+        # column length. So iterate over cells and find index for the first blank value
+        max_columns = self.ws.get_highest_column()
+        first_empty_index = max_columns
+        for i in range(1, max_columns + 1):
+            if not self.ws.cell(row=1, column=i).value:
+                first_empty_index = i
+                break
+        for i in range(0, self.max_attempts_in_sheet):
+            column_num = first_empty_index + i
+            self.ws.cell(row=1, column=column_num).value = "Attempt {index}".format(index=i + 1)
+
+    def _save_file(self, repeater_id, state):
+        if self.max_attempts_in_sheet > 0:
+            self._add_header_for_attempts()
+        file_name = "repeater_report_{repeater_id}_{state}_{timestamp}.xlsx".format(
+            repeater_id=repeater_id if repeater_id else '',
+            state=state if state else '',
+            timestamp=datetime.now().strftime("%Y-%m-%d-%H-%M-%S"),
+        )
+        self.wb.save(file_name)
+        return file_name
+
+    def _load_record_ids_from_file(self, file_path):
+        with open(file_path, 'rU') as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                self.record_ids.append(row['record_id'])
+
+    def handle(self, *args, **options):
+        domain = options.get('domain')
+        repeater_id = options.get('repeater_id')
+        state = options.get('state')
+        records_file_path = options.get('records_file_path')
+
+        if records_file_path:
+            self._load_record_ids_from_file(records_file_path)
+            records = self.record_ids
+            record_count = len(records)
+        elif domain and repeater_id:
+            if state:
+                records = iter_repeat_records_by_domain(domain, repeater_id=repeater_id, state=state)
+                record_count = get_repeat_record_count(domain, repeater_id=repeater_id, state=state)
```

Comment:
I think both of these methods work if `state` is `None`, so you could remove this if statement